mlrun/model_monitoring/db/tsdb/v3io/v3io.py

```diff
@@ -0,0 +1,365 @@
+        self,
+        project: str,
+        access_key: str = None,
+        container: str = "users",
+        v3io_framesd: str = None,
+        create_table: bool = False,
+    ):
+        super().__init__(project=project)
+        self.access_key = access_key or mlrun.mlconf.get_v3io_access_key()
+
+        self.container = container
+
+        self.v3io_framesd = v3io_framesd or mlrun.mlconf.v3io_framesd
+        self._frames_client: v3io_frames.client.ClientBase = (
+            self._get_v3io_frames_client(self.container)
+        )
+        self._v3io_client: V3IOClient = mlrun.utils.v3io_clients.get_v3io_client(
+            endpoint=mlrun.mlconf.v3io_api,
+        )
+
+        self._init_tables_path()
+
+        if create_table:
+            self.create_tsdb_application_tables()
+
+    def _init_tables_path(self):
+        self.tables = {}
+
+        events_table_full_path = mlrun.mlconf.get_model_monitoring_file_target_path(
+            project=self.project,
+            kind=mm_constants.FileTargetKind.EVENTS,
+        )
+        (
+            _,
+            _,
+            events_path,
+        ) = mlrun.common.model_monitoring.helpers.parse_model_endpoint_store_prefix(
+            events_table_full_path
+        )
+        self.tables[mm_constants.V3IOTSDBTables.EVENTS] = events_path
+
+        monitoring_application_full_path = (
+            mlrun.mlconf.get_model_monitoring_file_target_path(
+                project=self.project,
+                kind=mm_constants.FileTargetKind.MONITORING_APPLICATION,
+            )
+        )
+        (
+            _,
+            _,
+            monitoring_application_path,
+        ) = mlrun.common.model_monitoring.helpers.parse_model_endpoint_store_prefix(
+            monitoring_application_full_path
+        )
+        self.tables[mm_constants.V3IOTSDBTables.APP_RESULTS] = (
+            monitoring_application_path + mm_constants.V3IOTSDBTables.APP_RESULTS
+        )
+        self.tables[mm_constants.V3IOTSDBTables.METRICS] = (
+            monitoring_application_path + mm_constants.V3IOTSDBTables.METRICS
+        )
+
+    def create_tsdb_application_tables(self):
+        """
+        Create the application tables using the TSDB connector. At the moment we support 2 types of application tables:
+        - app_results: a detailed result that includes status, kind, extra data, etc.
+        - metrics: a basic key value that represents a single numeric metric.
+        """
+        application_tables = [
+            mm_constants.V3IOTSDBTables.APP_RESULTS,
+            mm_constants.V3IOTSDBTables.METRICS,
+        ]
+        for table in application_tables:
+            logger.info("Creating table in V3IO TSDB", table=table)
+            self._frames_client.create(
+                backend=_TSDB_BE,
+                table=self.tables[table],
+                if_exists=IGNORE,
+                rate=_TSDB_RATE,
+            )
+
+    def apply_monitoring_stream_steps(
```

Comment:
Doesn't this function need to be broken up? It seems to contain some steps that are not related to V3IO.

Reply:
I kind of expect the abstraction to be at the `apply_tsdb_target` level.

Reply:
I'm not sure we are going to repeat these steps exactly in other TSDBs. Therefore, I preferred to keep it on the V3IO TSDB level at the moment.
 If for example we'll see that for TDEngine we need to do the exact logic then I'll move the "none-V3IO" steps into a higher level.